src/poetry/repositories/pypi_repository.py

```diff
@@ -72,18 +71,11 @@ def __init__(
         )
 
         self._cache_control_cache = FileCache(str(release_cache_dir / "_http"))
-        self._session = CacheControl(
```

Comment:
I am not a 100% convinced this is the right solution for this. I'd much rather have e-tag used in the cache key.

Skipping cache for empty responses or invalid cache values, should be handled by gracefully handling failure - ie. clear the invalid cache and retry or fallback.

Reply:
I guess that's the naive approach [here](https://github.com/python-poetry/poetry/pull/4717).

Reply:
That approach is at the request level and not really at the cache level. I think an intermediate approach (e.g. layered on top of CacheControl and factoring in ETAGS) is the way to go. The caching is an important optimization, and adding a retry system at the high-level HTTP (e.g. `.get()`) API seems more a hack than a fix.